### ISSUE TITLE
Fatal expression with custom message compile error

### ISSUE THREAD

**Issue by jbbjarnason** [boost-ext/ut#627]
## Expected Behavior
Stream message to newly incorporated fatal expression as per docs.

## Actual Behavior
Seems like the streaming operator is not implemented, 

```bash
<source>: In function 'int main()':
<source>:5:26: error: no match for 'operator<<' (operand types are 'boost::ext::ut::v2_0_1::detail::fatal_<boost::ext::ut::v2_0_1::detail::eq_<int, boost::ext::ut::v2_0_1::detail::integral_constant<2> > >' and 'const char [16]')
    5 |   expect(fatal(1 == 2_i) << "fatal assertion");
      |          ~~~~~~~~~~~~~~~ ^~ ~~~~~~~~~~~~~~~~~
      |               |             |
      |               |             const char [16]
      |               boost::ext::ut::v2_0_1::detail::fatal_<boost::ext::ut::v2_0_1::detail::eq_<int, boost::ext::ut::v2_0_1::detail::integral_constant<2> > >
Compiler returned: 1
```

## Steps to Reproduce the Problem

  1. https://godbolt.org/z/WMeeWbEWY
```cpp
#include <https://raw.githubusercontent.com/boost-experimental/ut/master/include/boost/ut.hpp>

int main() {
  using namespace boost::ut;
  expect(fatal(1 == 2_i) << "fatal assertion");
  expect(1_i == 2);
}
```

## Specifications

  - Version: Master
  - Platform: Linux
  - Subsystem: Arch

**Comment by jbbjarnason** [comment]
This is a misspelling fixed in https://github.com/boost-ext/ut/pull/628 however the feature seems to be broken in some way, I have made a separate issue here: https://github.com/boost-ext/ut/issues/629

**Comment by jbbjarnason** [comment]
fixed by #628
